server/segment_rpc.go

```diff
@@ -65,7 +65,7 @@ func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
 	if paymentError := orch.ProcessPayment(payment, segData.ManifestID); paymentError != nil {
 
 		acceptableErr, ok := paymentError.(core.AcceptableError)
-		if ok && !acceptableErr.Acceptable() {
+		if !ok || !acceptableErr.Acceptable() {
```

Comment:
Do the `ProcessPayment()` unit tests need to be updated to cover this change?

Reply:
I don't think so as the output of this only changes for one error (TicketParams being nil).

Reply:
Misspoke in my first comment - I was actually thinking about the `ServeSegment()` unit tests and how they handle errors returned from `ProcessPayment()`.

At the moment, the [unacceptable ProcessPayment() error](https://github.com/livepeer/go-livepeer/blob/master/server/segment_rpc_test.go#L356) unit test mocks an error that adheres to the `core.AcceptableError` interface and returns false for `Acceptable()`. A useful addition to that test could be to mock an error that *does not* adhere to the `core.AcceptableError` interface.

Reply:
Fixed in 18b4192